### PR TITLE
Disallow hqImport in requirejs modules

### PR DESCRIPTION
**Pull request by orangejenny**
This is something I sometimes forget when migrating pages.

@jmtroth0 / @sravfeyn

### CODE REVIEW

**Comment by orangejenny** [comment]
@sravfeyn / @jmtroth0 can one of you review?

File: corehq/apps/hqwebapp/tests/test_requirejs.py [hunk 1]
```diff
@@ -9,21 +9,33 @@
 
 class TestRequireJS(SimpleTestCase):
 
-    def _get_hqdefine_files(self):
+    def setUp(self):
```
Comment (jmtroth0) [inline corehq/apps/hqwebapp/tests/test_requirejs.py]: `setUpClass` would probably work here too

File: corehq/apps/hqwebapp/tests/test_requirejs.py [hunk 2]
```diff
@@ -35,3 +47,26 @@ def test_files_match_modules(self):
...
+        errors = []
+        for filename in self.requirejs_files:
+            # Special case 1: ignore standard_hq_report.js until we migrate UCRs and reports
+            if filename.endswith("reports/js/standard_hq_report.js"):
+                continue
+            # Special case 2: knockout_bindings should be broken up, in the meantime, ignore
+            if filename.endswith("hqwebapp/js/knockout_bindings.ko.js"):
+                continue
+
+            proc = subprocess.Popen(["grep", "hqImport", filename], stdout=subprocess.PIPE)
```
Comment (jmtroth0) [inline corehq/apps/hqwebapp/tests/test_requirejs.py]: Might it be worth spinning the proc off into a greenlet pool or something so it isn't blocking

File: corehq/apps/hqwebapp/tests/test_requirejs.py [hunk 3]
```diff
@@ -35,3 +47,26 @@ def test_files_match_modules(self):
...
+            (out, err) = proc.communicate()
+            for line in out.split("\n"):
+                if line:
+                    match = re.search(r'hqImport\([\'"]([^\'"]*)[\'"]', line)
+                    if match:
+                        errors.append("{} imported in {}".format(match.group(1), filename))
```
Comment (jmtroth0) [inline corehq/apps/hqwebapp/tests/test_requirejs.py]: Cool use of groups!

**Review by jmtroth0 — APPROVED** [review]
This seems like it could add a bunch of time to each test run. I definitely agree it's worth it, but i'm curious about how much? If I'm totally off base feel free to ignore

File: corehq/apps/hqwebapp/tests/test_requirejs.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqwebapp/tests/test_requirejs.py]: https://github.com/dimagi/commcare-hq/pull/20380/commits/0491f5b62adae04f8c495c512e5f66100340fbef

**Comment by orangejenny** [comment]
Locally, the setup takes 5 sec and the new test itself is half a second, which seems not bad(?).

**Comment by jmtroth0** [comment]
Yeah, that seems fine for a test without the repeated setup

File: corehq/apps/hqwebapp/tests/test_requirejs.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/hqwebapp/tests/test_requirejs.py]: 720f2bd

**Pull request merged**